src/editor/nodes/FloorPlanNode.js

```diff
@@ -52,6 +60,20 @@ export default class FloorPlanNode extends EditorNodeMixin(FloorPlan) {
     node.maxTriangles = maxTriangles || 1000;
     node.forceTrimesh = forceTrimesh || false;
 
+    node._navMeshMode = navMeshMode || NavMeshMode.Automatic;
+
+    if (navMeshSrc) {
+      if (navMeshMode === NavMeshMode.Custom) {
+        loadAsync(
+          (async () => {
+            await node.load(navMeshSrc, onError);
+          })()
+        );
+      } else {
+        node._navMeshSrc = navMeshSrc;
```

Comment:
You might consider just assigning this above right after navMeshMode like:
`node._navMeshSrc = navMeshSrc || ""`

to get rid of the need for the nested if, and make the default value more explicit.